src/core/control/tools/StrokeHandler.cpp

```diff
@@ -43,13 +44,23 @@ void StrokeHandler::draw(cairo_t* cr) {
         cairo_set_operator(crMask, CAIRO_OPERATOR_CLEAR);
         cairo_paint(crMask);
 
-        cairo_set_operator(crMask, CAIRO_OPERATOR_SOURCE);
-        view.drawStroke(crMask, stroke, true);
+        xoj::View::Context context{crMask, xoj::View::NORMAL_NON_AUDIO, xoj::View::HIDE_CURRENT_EDITING,
+                                   xoj::View::COLORBLIND};
+        xoj::View::StrokeView strokeView(stroke);
+        strokeView.draw(context);
     }
-    DocumentView::applyColor(cr, stroke);
 
-    cairo_set_operator(
-            cr, stroke->getToolType() == STROKE_TOOL_HIGHLIGHTER ? CAIRO_OPERATOR_MULTIPLY : CAIRO_OPERATOR_OVER);
+    if (stroke->getToolType() == STROKE_TOOL_HIGHLIGHTER) {
```

Comment:
I think a comment here to explain the code would be helpful:
```suggestion
    // set the stroke color and blend mode
    if (stroke->getToolType() == STROKE_TOOL_HIGHLIGHTER) {
```